File: src/ArcadeMode/island_of_knowledge/BoxBlur.java
package ArcadeMode.island_of_knowledge;

import java.util.ArrayList;
import java.util.List;

public class BoxBlur {
  public static void main(String[] args){
    int[][] test = new int[4][4];
    test[0][0]= 7;
    test[0][1]= 4;
    test[0][2]= 0;
    test[0][3]= 1;
    test[1][0]= 5;
    test[1][1]= 6;
    test[1][2]= 2;
    test[1][3]= 2;
    test[2][0]= 6;
    test[2][1]= 10;
    test[2][2]= 7;
    test[2][3]= 8;
    test[3][0]= 1;
    test[3][1]= 4;
    test[3][2]= 2;
    test[3][3]= 0;

    boxBlur(test);
  }

  public static int[][] boxBlur(int[][] image){
    List<Integer> res = new ArrayList<Integer>();
      for(int i = 0; i<image.length;++i){
        for(int j  =0; j<image[i].length;++j){

        }
      }
    return null;
    }
}

File: src/ArcadeMode/island_of_knowledge/BoxBlur.kt
package ArcadeMode.island_of_knowledge


fun main(args: Array<String>) {
    print(boxBlur(mutableListOf(mutableListOf(7, 4, 0, 1),
            mutableListOf(5, 6, 2, 2),
            mutableListOf(6, 10, 7, 8),
            mutableListOf(1, 4, 2, 0))))
}

fun boxBlur(image: MutableList<MutableList<Int>>): MutableList<MutableList<Int>> {
    var upper = 0
    var down = 0
    val blurArray = MutableList(image.size - 2) { MutableList(image[0].size - 2) { 0 } }

    while ((image.size) - upper >= 3 &&(image[0].size) - down >= 3) {
        var result = 0

        for (i in 0 until 3) {
            for (j in 0 until 3) {

                result += image[upper + i][down + j]
            }
        }
        blurArray[upper][down] = (result / 9)
        if ((image[0].size - 1) - down >= 3) {
            down++
        } else {
            upper++
            down = 0
        }


    }
    return blurArray
}